.gitlab/docker/Dockerfile.spack_compiler

```diff
@@ -0,0 +1,32 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+ARG ARCH
+ARG BOOST
+ARG BUILD_TYPE
+ARG COMPILER
+ARG CXXSTD
+ARG HWLOC
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+SHELL ["/bin/bash", "-c"]
+
+# Disable host compatibility to be able to compile for broadwell on zen2
+RUN echo -e "concretizer:\n  targets:\n    host_compatible: false" \
+    > $SPACK_ROOT/etc/spack/concretizer.yaml
+
+# Install compiler
+RUN export spack_compiler=${COMPILER/clang/llvm} && \
+    spack install $spack_compiler languages=c++ arch=$ARCH && \
+    spack compiler add --scope site $(spack location -i $spack_compiler)
+
+ENV spack_spec "pika@main build_type=$BUILD_TYPE arch=$ARCH %$COMPILER malloc=system cxxstd=$CXXSTD ^$BOOST ^$HWLOC"
```

Comment:
I would leave out the build type at this stage. All the dependencies are built with a build type independent of the pika build type so if we leave it out here we can use the same dependencies images for release and debug builds.